src/App.jsx

```diff
@@ -14,45 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
+
+function getSortedGoods(goods, sortField, reversed) {
+  let sortedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_ALPHABETICALLY:
+        sortedGoods = sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SORT_BY_LENGTH:
+        sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        return 0;
+    }
+  }
+
+  if (reversed) {
+    return sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortField, setSortField] = useState('');
+
+  const [visibleGoods, setVisibleGoods] = useState(
+    getSortedGoods(
+      goodsFromServer, sortField, reversed,
+    ),
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function handleReverse() {
+    setReversed(!reversed);
+    setVisibleGoods(getSortedGoods(goodsFromServer, sortField, !reversed));
+  }
+
+  function handleSort(field) {
+    setSortField(field);
+    setVisibleGoods(getSortedGoods(goodsFromServer, field, reversed));
+  }
```

Comment:
don't see any reason to call setVisibleGoods here, because after every state changes it will be executed on the 52
check other places 
```suggestion
  function handleSort(field) {
    setSortField(field);
  }
```